shell/browser/net/proxying_url_loader_factory.cc

```diff
@@ -326,6 +326,14 @@ bool ProxyingURLLoaderFactory::IsForServiceWorkerScript() const {
 
 void ProxyingURLLoaderFactory::InProgressRequest::OnLoaderCreated(
     mojo::PendingReceiver<network::mojom::TrustedHeaderClient> receiver) {
+  // When CORS is involved there may be multiple network::URLLoader associated
+  // with this InProgressRequest, because CorsURLLoader may create a new
+  // network::URLLoader for the same request id in redirect handling - see
+  // CorsURLLoader::FollowRedirect. In such a case the old network::URLLoader
+  // is going to be detached fairly soon, so we don't need to take care of it.
+  // We need this explicit reset to avoid a DCHECK failure in mojo::Receiver.
+  header_client_receiver_.reset();
```

Comment:
Is this `.reset()` needed given we immediately `.Bind()` on the next line?

Reply:
This is a cherry-pick of https://chromium-review.googlesource.com/c/chromium/src/+/2011781. Reason for reset is stated above in comment by original author, it is here to prevent DCHECK in mojo receiver.